src/types.ts

```diff
@@ -81,6 +81,18 @@ export interface ArweaveWalletConnector {
   disconnect(): Promise<void>;
   getWalletAddress(): Promise<string>;
   getWalletBalanceAR(): Promise<string>;
+  getWalletANTs(
+    approvedSourceCodeTransactions: string[],
+    cursor?: string,
+  ): Promise<{ ids: string[]; cursor?: string }>;
```

Comment:
```suggestion
```

Reply:
we also should update the type of the global state so the wallet is:
```
export type GlobalState = {
  arweave: Arweave;
  arnsSourceContract: ArNSContractState;
  gateway: string;
  walletAddress?: string;
  wallet?: ArweaveWalletConnector & ArweaveGraphQLAPI;
  arnsContractId: string;
  showConnectWallet: boolean;
  errors: Array<Error>;
  notifications: { id: string; text: string }[];
};
```
and the reducer
```
export type Action =
  | { type: 'setWalletAddress'; payload: ArweaveTransactionId | undefined }
  | {
      type: 'setWallet';
      payload: (ArweaveWalletConnector & ArweaveGraphQLAPI) | undefined;
    }
  | { type: 'setGateway'; payload: string }
  | { type: 'setArnsContractState'; payload: ArNSContractState }
  | { type: 'setShowConnectWallet'; payload: boolean }
  | { type: 'setArweave'; payload: Arweave }
  | { type: 'pushNotification'; payload: string }
  | { type: 'removeNotification'; payload: string };
```